genesis/utils/path_planing.py

```diff
@@ -0,0 +1,1127 @@
+                    for i_q in range(self.n_qs):
+                        self._solver.qpos[i_q + self._q_start, i_b] = steer_result[i_q]
+                    self._solver._func_forward_kinematics_entity(self._entity._idx_in_solver, i_b)
+                    self._solver._func_update_geoms(i_b)
+
+    @ti.kernel
+    def _kernel_rrt_step2(
+        self,
+        ignore_geom_pairs: ti.types.ndarray(),
+        envs_idx: ti.types.ndarray(),
+    ):
+        ti.loop_config(serialize=self._solver._para_level < gs.PARA_LEVEL.ALL)
+        for i_b in envs_idx:
+            if self._rrt_is_active[i_b]:
+
+                is_collision_detected = self._func_check_collision(ignore_geom_pairs, i_b)
+                if is_collision_detected:
+                    self._rrt_tree_size[i_b] -= 1
+                    self._rrt_node_info[self._rrt_tree_size[i_b], i_b].configuration = 0.0
+                    self._rrt_node_info[self._rrt_tree_size[i_b], i_b].parent_idx = -1
+                else:
+                    # check the obtained steer result is within goal configuration only if no collision
+                    is_valid = True
+                    for i_q in range(self.n_qs):
+                        if (
+                            abs(self._solver.qpos[i_q + self._q_start, i_b] - self._rrt_goal_configuration[i_q, i_b])
+                            > self._rrt_pos_tol
+                        ):
+                            is_valid = False
+                            break
+                    if is_valid:
+                        self._rrt_goal_reached_node_idx[i_b] = self._rrt_tree_size[i_b] - 1
+                        self._rrt_is_active[i_b] = 0
+
+    def plan(
+        self,
+        qpos_goal,
+        qpos_start=None,
+        *,
+        resolution=0.01,
+        smooth_path=True,
+        num_waypoints=100,
+        envs_idx=None,
+    ):
+        """
+        Plan a path from `qpos_start` to `qpos_goal`.
+        qpos_goal : array_like
+            The goal state. [B, Nq] or [1, Nq]
+        qpos_start : None | array_like, optional
+            The start state. If None, the current state of the rigid entity will be used. Defaults to None. [B, Nq] or [1, Nq]
+        """
+        import time
+
+        if self._solver.n_envs > 0:
+            envs_idx = self._solver._sanitize_envs_idx(envs_idx)
+        else:
+            envs_idx = torch.zeros(1, dtype=gs.tc_int, device=gs.device)
+
+        qpos_cur = self._entity.get_qpos()
+        qpos_goal, qpos_start = self.validate_input_qpos_batch(qpos_goal, qpos_start, envs_idx)
+
+        self._init_rrt_fields()
+        self._reset_rrt_fields()
+        self._kernel_rrt_init(qpos_start.contiguous(), qpos_goal.contiguous(), envs_idx)
+
+        self._entity.set_qpos(qpos_start)
+        self._solver._kernel_detect_collision()
+        scene_contact_info = self._solver.collider.contact_data.to_torch(gs.device)
+        n_contacts = self._solver.collider.n_contacts.to_torch(gs.device)
```

Comment:
Do not use `self._solver.collider.n_contacts.to_torch(gs.device)` but rather the new API.